app/src/main/java/com/zeronfinity/cpfy/MainActivity.kt

```diff
@@ -1,11 +1,78 @@
 package com.zeronfinity.cpfy
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import java.net.HttpURLConnection
+import java.net.URL
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+import kotlin.coroutines.CoroutineContext
+
+class MainActivity: AppCompatActivity(), CoroutineScope {
+    lateinit var job: Job
+
+    override val coroutineContext: CoroutineContext
+        get() = job + Dispatchers.Main
+
+    val contestApiUrl = "https://clist.by/api/v1/json/contest/"
+    val apiUsername = "Zeronfinity"
+    val apiKey = BuildConfig.CLIST_API_KEY
+    val apiDateFormat = "yyyy-MM-dd'T'HH:mm:ss"
+    val numberOfDaysForApiEndDate = 7
+
+    data class ContestData(val name: String, val duration: Int, val platformName: String, val startTime: String, val endTime: String, val url: String)
```

Comment:
Start time and end time, should it be string? Isn't there `DateTime' or something equivalent?

Reply:
The data type will be updated in a later change when these values are used, based on the necessity then, so string is like a placeholder here for now.